README.md

```diff
@@ -1,285 +1,366 @@
 
-	**Errors:**
-	- _`payToken.transferFrom`_ -  might cause the TX to fail if the caller hadn't authorized `this` contract to do transfers.
-	- _`otc.sellAllAmount`_ - might cause the TX to fail if the `buyAmt` results being less than `minBuyAmt`
-	- _`withdrawAndSend`_ - might cause the TX to fail
+**Summary:** The ProxyCreateAndExcecute contract is used to perform direct/instant trading by leveraging the existing Maker OTC platform. In this case, the caller of the methods is interested in creating a `DSProxy` that can be reused across other Maker Products as well as for selling and/or buying an exact amount of a given token for another token.
 
-	---
+This contract works by extending the **OasisDirectProxy,** plus some additional methods. Each of the additional methods described below has the same functionality (calls two methods). 
 
-- **`buyAllAmount`** - Used to exchange `ERC-20 token` for `ERC-20 token`. In the beginning the amount to be paid is calculated. `payAmt` is transferred to `this` contract.  After that the `otc` contract is ensured to have enough allowance in the `payToken` to do transfers on the behalf of `this` contract. Follows buy operation where `buyAmt` is bought. Last step includes transfer of the bought amount to the caller.
+One of the main call methods is the `DSProxyFactor.build` . This method creates a new proxy for a sender address. The other call in each of the additional methods is used to retrieve the amount that will be `sold` or `bought`.
 
-	**Arguments:**
-	- _[OtcInterface](https://github.com/makerdao/maker-otc-support-methods/blob/master/src/MakerOtcSupportMethods.sol#L5) `otc`_ - An address of the contract that represents the OTC market contract. Only a few methods are exposed.
-	 - _[TokenInterface](https://github.com/makerdao/oasis-direct-proxy/blob/master/src/OasisDirectProxy.sol#L11) `buyToken`_ - An address of any `ERC-20 token` or token that implements the interface
-	 - _`uint buyAmt`_ - The amount that will be bought
-	 - _[TokenInterface](https://github.com/makerdao/oasis-direct-proxy/blob/master/src/OasisDirectProxy.sol#L11) `payToken`_ - An address of any `ERC-20 token` or token that implements the interface
-	 - _`uint maxPayAmt`_ - please check _*threshold_
+### threshold
 
-	**Returns:**
-	- _`uint payAmt`_ -the amount of `payToken` that will be paid
+In order to understand exactly what `threshold` does, let's go though a quick example. 
 
-	**Errors:**
-	- _`otc.getPayAmount`_ - might cause the TX to fail if there aren't enough order to fill the `buyAmt`
-	- might cause the TX to fail if the calculated amount to be paid is higher than `maxPayAmt`
-	- _`payToken.transferFrom`_ - might cause the TX to fail
-	- _`otc.buyAllAmount`_ - might cause the TX to fail if we have to pay more than the calculated amount to be paid in the beginning of the TX.
-	- _`buyToken.transfer`_ - might cause the TX to fail
+**Example:** Let's say that someone wants to sell `10 MKR` at the price of `1.5 MKR/DAI`. 
```

Comment:
We can leave this example in here as well but  those kind of examples will be done for the OTC `buyAllAmount` and `sellAllAmount`